### PR TITLE
[5.0] Adding plural hasColumn method hasColumns

### PR DESCRIPTION
**Pull request by christophrumpel**
As approved in #5015 I added a plural hasColumns method to check an if an array of columns exist in a certain database table.

### CODE REVIEW

File: src/Illuminate/Database/Schema/Builder.php [hunk 1]
```diff
@@ -67,7 +67,30 @@ public function hasColumn($table, $column)
...
+    public function hasColumns($table, $columns)
+    {
+        $tableColumns = $this->getColumnListing($table);
+        array_map('strtolower', $tableColumns);
+
+        foreach ($columns as $column)
+        {
+            $column = strtolower($column);
+            if ( ! in_array($column, $tableColumns))
+                return false;
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Schema/Builder.php]: this needs to be on the same line as the logic

File: src/Illuminate/Database/Schema/Builder.php [hunk 2]
```diff
@@ -67,7 +67,30 @@ public function hasColumn($table, $column)
...
 
-	/**
+    /**
+     * Determine if the given table has given columns.
+     *
+     * @param  string  $table
+     * @param  array   $columns
+     * @return bool
+     */
+    public function hasColumns($table, $columns)
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Schema/Builder.php]: Please fix your indentation to use tabs.

Comment (GrahamCampbell) [inline src/Illuminate/Database/Schema/Builder.php]: Also, you're missing a typehint there.

File: src/Illuminate/Database/Schema/Builder.php [hunk 3]
```diff
@@ -67,7 +67,30 @@ public function hasColumn($table, $column)
...
+        {
+            $column = strtolower($column);
+            if ( ! in_array($column, $tableColumns))
+                return false;
+        }
+
+        return true;
+    }
+
+
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Schema/Builder.php]: one a single space between methods please

File: src/Illuminate/Database/Schema/Builder.php [hunk 4]
```diff
@@ -67,7 +67,30 @@ public function hasColumn($table, $column)
...
+     * @return bool
+     */
+    public function hasColumns($table, $columns)
+    {
+        $tableColumns = $this->getColumnListing($table);
+        array_map('strtolower', $tableColumns);
+
+        foreach ($columns as $column)
+        {
+            $column = strtolower($column);
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Schema/Builder.php]: no need for this extra variable asignment

**Comment by christophrumpel** [comment]
Thanks GrahamCampbell, strange I didn't see the indent problems in PhpStorm. You also mentioned a typehint is missing. Where would that be?

**Comment by GrahamCampbell** [comment]
$columns needs an array typehint

File: src/Illuminate/Database/Schema/Builder.php [hunk 5]
```diff
@@ -67,7 +67,29 @@ public function hasColumn($table, $column)
 		return in_array($column, array_map('strtolower', $this->getColumnListing($table)));
 	}
 
-	/**
+    	/**
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Schema/Builder.php]: not indeted right still

File: src/Illuminate/Database/Schema/Builder.php [hunk 6]
```diff
@@ -67,7 +67,29 @@ public function hasColumn($table, $column)
...
+    	public function hasColumns($table, $columns)
+    	{
+        	$tableColumns = $this->getColumnListing($table);
+        	
+        	array_map('strtolower', $tableColumns);
+
+        	foreach ($columns as $column)
+        	{
+            		if ( ! in_array(strtolower($column), $tableColumns))
+                		return false;
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Schema/Builder.php]: no. this MUST go on the same line, or you MUST use braces. Code without that is asking for trouble.

File: tests/Database/DatabaseSchemaBuilderTest.php [hunk 7]
```diff
@@ -24,4 +24,16 @@ public function testHasTableCorrectlyCallsGrammar()
 		$this->assertTrue($builder->hasTable('table'));
 	}
 
+    public function testTableHasColumns()
```
Comment (GrahamCampbell) [inline tests/Database/DatabaseSchemaBuilderTest.php]: test methods must be separated by exactly 2 blank lines.

**Comment by christophrumpel** [comment]
Yeah right, thanks. I added the typehint and added another blank line between the tests.

File: tests/Database/DatabaseSchemaBuilderTest.php [hunk 8]
```diff
@@ -24,4 +24,17 @@ public function testHasTableCorrectlyCallsGrammar()
 		$this->assertTrue($builder->hasTable('table'));
 	}
 
+
```
Comment (GrahamCampbell) [inline tests/Database/DatabaseSchemaBuilderTest.php]: remove trailing whitespace please

**Comment by GrahamCampbell** [comment]
One trailing tab character to remove, then please squash to one commit.

**Comment by christophrumpel** [comment]
ah damn, I srewed the squashing.

**Comment by christophrumpel** [comment]
Ok I did a undo of the rebase and rebased it again. I hope everything is ok now.

**Comment by GrahamCampbell** [comment]
Why have you modified the entire file?

**Comment by christophrumpel** [comment]
That happenend during my wrong rebase attempt. I will try to fix that, sry.

**Comment by christophrumpel** [comment]
I don't know why the complete file changed. How can I change that? Can I copy the current 5.0 version of the file in the commit file and add my function? Or will that change the complete file again?

**Comment by franzliedke** [comment]
@christophrumpel Your editor should be able to turn the indentation back to tabs, that should "revert" the changes.

**Comment by christophrumpel** [comment]
Thx @franzliedke , so I think I got that now. Now I just need to merge my commits to one and then only my added function should be changed in comparison to the current 5.0 version right?

**Comment by christophrumpel** [comment]
Great, thx everybody for the assistance!

File: src/Illuminate/Database/Schema/Builder.php [hunk 9]
```diff
@@ -68,6 +68,27 @@ public function hasColumn($table, $column)
...
+	 *
+	 * @param  string  $table
+	 * @param  array   $columns
+	 * @return bool
+	 */
+	public function hasColumns($table, array $columns)
+	{
+		$tableColumns = $this->getColumnListing($table);
+
+		array_map('strtolower', $tableColumns);
```
Comment (JosephSilber) [inline src/Illuminate/Database/Schema/Builder.php]: Doesn't `array_map` return a new array?

Comment (taylorotwell) [inline src/Illuminate/Database/Schema/Builder.php]: yeah, that looks broken. Can someone fix?

**Comment by christophrumpel** [comment]
Oh thx, I will. Can I create a new commit to the same branch?

**Comment by franzliedke** [comment]
That should work. You'll still have to create a new pull request from that branch, though,.

**Comment by taylorotwell** [comment]
Open a new PR to fix it.

**Comment by christophrumpel** [comment]
Ok thx, I will do that!

File: src/Illuminate/Database/Schema/Builder.php (hunk 9 above)
Comment (JosephSilber) [inline src/Illuminate/Database/Schema/Builder.php]: @taylorotwell #8355

**Pull request merged**